pkg/monitoring/metrics/virt-controller/vmdisk_metrics.go

```diff
@@ -0,0 +1,103 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright the KubeVirt Authors.
+ */
+
+package virt_controller
+
+import (
+	"context"
+	"errors"
+
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+	io_prometheus_client "github.com/prometheus/client_model/go"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+)
+
+var (
+	vmdiskMetrics = []operatormetrics.Metric{
+		vmAllocatedDiskSize,
+		vmUsedDiskSize,
+	}
+
+	vmAllocatedDiskSize = operatormetrics.NewGaugeVec(
+		operatormetrics.MetricOpts{
+			Name: "kubevirt_vm_disk_allocated_size_bytes",
+			Help: "Disks allocated size in bytes.",
+		},
+		[]string{"namespace", "vm_name", "disk_name", "pvc_name"}, // Updated labels
+	)
+
+	vmUsedDiskSize = operatormetrics.NewGaugeVec(
+		operatormetrics.MetricOpts{
+			Name: "kubevirt_vm_disk_used_size_bytes",
+			Help: "Disks used size in bytes.",
+		},
+		[]string{"namespace", "vm_name", "disk_name", "pvc_name"}, // Updated labels
+	)
+)
+
+func SetVmAllocatedDiskSize(vm *v1.VirtualMachine, diskName string) error {
+	clientset, err := kubecli.GetKubevirtClient()
+	if err != nil {
+		return err
+	}
+
+	for _, volume := range vm.Spec.Template.Spec.Volumes {
+		if volume.Name == diskName && volume.PersistentVolumeClaim != nil {
+			pvcName := volume.PersistentVolumeClaim.ClaimName
+			pvc, err := clientset.CoreV1().PersistentVolumeClaims(vm.Namespace).Get(context.TODO(), pvcName, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+
+			pvName := pvc.Spec.VolumeName
+			if pvName == "" {
+				return errors.New("no PersistentVolume bound to PVC")
+			}
+
+			pv, err := clientset.CoreV1().PersistentVolumes().Get(context.TODO(), pvName, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+
+			if capacity, ok := pv.Spec.Capacity[k8sv1.ResourceStorage]; ok {
+				sizeInBytes := capacity.Value()
```

Comment:
Using capacity is fine for block PVC but does not work for filesystem PVCs on shared storage (kubevirt hpp or local storage for example). For that case you have to get the size of the `disk.img` file on the PVC

Reply:
I thought that this metric covers the filesystem bytes https://github.com/kubevirt/kubevirt/blob/main/docs/metrics.md#kubevirt_vmi_filesystem_capacity_bytes, no?

Reply:
Those stats are from guest perspective. Also it is an aggregate.

Reply:
+1 so I will add also the filesystem PVCs thanks a lot!

Reply:
@mhenriks sorry for bothering again, do you have maybe a reference how to get the disk.img file size? struggling to understand